### PR TITLE
Update padding requirements to fix #8

### LINKED ISSUE #8: Padding around mutation doesn't always obey user's request
See here:

```
# even if user doesn't want any padding around the mutation we need
# to at least include enough of the surrounding non-mutated
# residues to construct candidate epitopes of the specified lengths
min_required_padding = max(self.epitope_lengths) - 1
padding_around_mutation = max(
  padding_around_mutation, min_required_padding)
```

I found this pretty confusing, and think an error should simply be thrown if a bad padding value is passed.

### PR DESCRIPTION
**Pull request by tavinathanson**
We now raise an error if invalid padding is specified: see #8

### CODE REVIEW

**Comment by arahuja** [comment]
My first look at this code but I'm confused as to what `padding_around_mutation` should be.  A few questions:
- it's padding in amino acid residues, not DNA?
- is `padding_around_mutation` adding an equal amount of bases to both sides and if so, is that  `padding_around_mutation` or  `padding_around_mutation/2` i.e. if `padding_around_mutation=None` it defaults to the epitope length, the padding is epitope_length/2 on both sides?

**Comment by tavinathanson** [comment]
I was also confused when I first looked at it, and this PR is an attempt to at least make it somewhat less confusing :)

`padding_around_mutation` is indeed amino acid residues. It's adding `padding_around_mutation` to both sides.

Example:

Let's say we have a mutation M:

XXXXXXXXXMXXXXXXXXX

All the possible 9mers look like:

XXXXXXXXXM
...
MXXXXXXXXX

So we need the entire 9 on both sides, for a total of 19, to send all possible 9mers through to mhctools.

**Comment by arahuja** [comment]
This PR itself looks good to me.  I'm a bit confused on what `min_required_padding` does and when it comes into play especially as it seems it would be different for frameshift vs MNV, but that's separate of this PR.

**Comment by tavinathanson** [comment]
@arahuja Yeah, I agree that's a bit confusing. Check out https://github.com/hammerlab/topiary/blob/master/topiary/epitope_helpers.py#L36

Basically, the padding is applicable to all mutations: it's added to the left of the start of the mutation, and to the right of the end of the mutation.

**Comment by tavinathanson** [comment]
(Err, I might be misunderstanding. One sec.)

**Comment by tavinathanson** [comment]
Yeah, I think I stand by the above. See https://github.com/hammerlab/varcode/blob/master/varcode/effects.py#L608, where `aa_mutation_end_offset` is set to start + shifted sequence length.

**Comment by iskandr** [comment]
@tavinathanson Is right in his interpretation. The reason the user has any control over this is because we need longer-than-epitope sequences for the downstream PGV code.

File: topiary/mutant_epitope_predictor.py [hunk 1]
```diff
@@ -96,12 +98,18 @@ def __init__(
...
-        padding_around_mutation = max(
-            padding_around_mutation, min_required_padding)
+        if padding_around_mutation is None:
+            padding_around_mutation = min_required_padding
+        else:
+            if padding_around_mutation < min_required_padding:
+                raise ValueError("Padding around mutation cannot "
+                                 "be less than %d for epitope lengths "
+                                 "%s" % (min_required_padding,
+                                         self.epitope_lengths))
```
Comment (iskandr) [inline topiary/mutant_epitope_predictor.py]: Is the error string missing a string formatter for the epitope lengths?

**Comment by iskandr** [comment]
LGTM

**Pull request merged**